Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1020,6 +1020,11 @@ - (void)resumeApp:(dispatch_block_t)completion {
             [self hideSplashView];
             done();
         }
+        WMFFundraisingCampaignDataController *controller = [WMFFundraisingCampaignDataController sharedInstance];
+        NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+        [controller setInstallIDProvider:^NSString * _Nonnull{
+            return defaults.wmf_appInstallId;
+        }];
```

Comment:
There's an `WMFDataEnvironment.current.appInstallIDUtility` closure [here](https://github.com/wikimedia/wikipedia-ios/blob/main/WMFData/Sources/WMFData/Environment/WMFDataEnvironment.swift#L24) that we're already using elsewhere that you should be able to call directly from `WMFFundraisingCampaignDataController`.